### ISSUE TITLE
The recent total-delay metrics are rounded up to seconds

### ISSUE THREAD

**Issue by xemul** [scylladb/scylla#10254]
There are two metrics added to seastar (and scylla) recently -- `io_queue_total_delay_sec` and `io_queue_total_exec_sec`. Both are counters (derives) and both are reporting accumulated durations in _seconds_, which are naturally floating point numbers. Unfortunately, seastar metrics engine converts all `DERIVE` metric values to integers:

```c++
static std::string to_str(const seastar::metrics::impl::metric_value& v) {
    switch (v.type()) {
    case seastar::metrics::impl::data_type::GAUGE:
        return std::to_string(v.d());
    case seastar::metrics::impl::data_type::COUNTER:
    case seastar::metrics::impl::data_type::ABSOLUTE:
        return std::to_string(v.ui());
    case seastar::metrics::impl::data_type::DERIVE:
        return std::to_string(v.i());    <----------------------------------------------- here
    case seastar::metrics::impl::data_type::HISTOGRAM:
        break;
    }
    return ""; // we should never get here but it makes the compiler happy
} 
...
class metric_value {
public: 
...
    int64_t i() const {
        ...
    }

```

As the result both metrics lose precision a lot -- they are reported as plain integers. When graphana applies `rate()` to them the result becomes stair-case function, with e.g. 25ms steps (#10064).

**Comment by xemul** [comment]
scylladb/seastar#446

**Comment by xemul** [comment]
Fixed by 043b1c7f89d0dac7b89985b03584b79d6d5d8eb7

**Comment by bhalevy** [comment]
@xemul backport candodates?

**Comment by xemul** [comment]
4.6.3 and 4.5.6 need this, but it depends on whether metrics with fixed scylladb/scylla-monitoring#1570 (AFAIU 3.10) supports those scyllas

**Comment by avikivity** [comment]
Fixed in 5.0 by a8eb5164b2727b8c146cc26289ac5d630587b0fe

**Comment by avikivity** [comment]
I suppose there's no need to disturb 4.6 for this.
